src/realm/object-store/property.hpp

```diff
@@ -336,6 +345,26 @@ inline Property::Property(std::string name, PropertyType type, std::string objec
 {
 }
 
+inline Property::Property(std::string name, PropertyType type, const NestedTypes& nested_types,
+                          std::string public_name)
+    : name(std::move(name))
+    , public_name(std::move(public_name))
+    , type(type)
+    , nested_types(nested_types)
+{
+}
+
+inline Property::Property(std::string name, PropertyType type, const NestedTypes& nested_types,
+                          std::string object_type, std::string link_origin_property_name, std::string public_name)
+    : name(std::move(name))
+    , public_name(std::move(public_name))
+    , type(type)
+    , object_type(std::move(object_type))
+    , link_origin_property_name(std::move(link_origin_property_name))
+    , nested_types(nested_types)
```

Comment:
this constructor appears to be untested, and I don't think it is correct to be setting the nested_types on a linking object property

Reply:
You can perfectly well have a list of linklist. Anyway - adding a test for it revealed that the two new constructors did not differ on the first 4 parameter types. That had to be fixed - perhaps not in a pretty way, but is there a better way?

Reply:
Thanks for the test on list of linklist. But I am still confused about why we would need to set `nested_types` on a linking object (ie PropertyType::LinkingObjects with a source `object_type` and `link_origin_property_name`). I suspect you don't need this new constructor at all. Or maybe I am not understanding how you expect to use it, can you could show how you intend to use it by adding a test?

Reply:
Ok - now I get it. link_origin_property_name is only needed for PropertyType::LinkingObjects.